### PR TITLE
In-App Paid User migration Flow UX [GROW-143]

### PR DESCRIPTION
**Pull request by gisete**
Display a modal to Legacy paid customers featuring the Essentials plan.  [GROW-143](https://buffer.atlassian.net/browse/GROW-143)
image

If the customer clicks on "Remind me Later" the modal will not show again for another day.
Clicking on "Learn more" will trigger the following modals with details about the plan and pricing.

image

image

image

### CODE REVIEW

**Review by federicoweber — COMMENTED** [review]
Great work on this one @gisete, I left some minor comments on the code, but overall this is looking good to me, I haven't left any comments on the UI itself, as @moreofmorris went over all of the issues already :).

I will add the logic on a separate PR, so that you can keep working on this one 🙂 , and we will consolidate it all, once ready before emerging it in.

File: packages/app-shell/src/components/Modal/modals/PaidMigration/EssentialsPlan/index.jsx [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+              />
+            </ButtonWrapper>
+          </BottomSection>
+        </Holder>
+      )}
+    </ModalContext.Consumer>
+  );
+};
+
+export const PlansFeatures = [
```
[the remark this answers is not in the document]
Comment (gisete) [reply packages/app-shell/src/components/Modal/modals/PaidMigration/EssentialsPlan/index.jsx:184]: Hey @federicoweber , quick question, did you mean I should move this into a new `config.js` file? I couldn't find an existing one. Thanks!

**Comment by gisete** [comment]
@federicoweber thanks for all the feedback, super helpful 🙌  I see what you're saying about the formatting changes making it harder to review the code, sorry about that! I removed all the changes made by Prettier if you want to quickly go over it again 🙂

File: packages/app-shell/src/components/Modal/modals/PaidMigration/EssentialsPricing/index.jsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+  if (migrationPreview?.currentPlan?.periodEnd) {
+    const date = new Date(migrationPreview.currentPlan.periodEnd);
+    return (<>
+      on <b>{months[date.getMonth()]} {date.getDate()}, {date.getFullYear()}</b>
+    </>)
+  }
+
+  return null
+}
```
Comment (federicoweber) [inline packages/app-shell/src/components/Modal/modals/PaidMigration/EssentialsPricing/index.jsx:56]: @gisete I've pushed this change that will allow us to properly display the period end date once we merge in https://github.com/bufferapp/core-authentication-service/pull/1158.

We will need to also request that in the schema, but we can't do that till the BE PR has been merged.

**Pull request merged**